File: 백준/Silver/15649. N과 M （1）/N과 M （1）.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let inputLines = [];
rl.on("line", (line) => {
  inputLines.push(line);
});

rl.on("close", () => {
  const [N, M] = inputLines[0].split(" ").map(Number);
  const arr = [];

  const dfs = () => {
    if (arr.length === M) {
      console.log(arr.join(" "));
      return;
    }

    for (let i = 1; i <= N; i++) {
      if (!arr.includes(i)) {
        arr.push(i);
        dfs();
        arr.pop();
      }
    }
  };

  dfs();
});

File: 백준/Silver/15649. N과 M （1）/N과 M （1）.py
import sys

input = sys.stdin.readline

N, M = map(int, input().split())

graph = []
for i in range(1, N + 1):
    graph.append(i)
visited = [0 for _ in range(N)]


def dfs(a):
    res.append(graph[a])

    if len(res) == M:
        print(*res)
        return

    for j in range(len(graph)):
        if visited[j] == 0:
            visited[j] = 1
            dfs(j)
            visited[j] = 0
            res.pop()


for i in range(N):
    res = []
    if visited[i] == 0:
        visited[i] = 1
        dfs(i)
        visited[i] = 0
